### PR TITLE
Create members components

### PR DESCRIPTION
**Pull request by Larkgronick**
Request date:
02/02/21

Total:
• Created members, members progress and members tasks pages
• Add drawer to navigate 
• Implemented Router for these pages
• Realized own design

### CODE REVIEW

File: src/components/Buttons/Button/Button.jsx [hunk 1]
```diff
@@ -1,17 +1,18 @@
 import PropTypes from 'prop-types';
-import noop from '../../../shared/noop';
+import './Button.css';
 
-export const Button = ({ children, onClick, ...restProps }) => (
-  <button type='button' onClick={onClick} {...restProps}>
-    {children}
-  </button>
-);
+export function Button(props) {
```
Comment (YauhenKoipish) [inline src/components/Buttons/Button/Button.jsx:4]: parse props here

File: src/components/Buttons/Hamburger/Hamburger.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export function Hamburger(props) {
+  let hamburgerClasses = 'bt-menu-trigger';
+  const { showDrawer, toggle } = props;
+
+  if (showDrawer) {
+    hamburgerClasses = 'bt-menu-trigger bt-menu-open';
+  }
+  return (
+    <button onClick={toggle} className={hamburgerClasses} type='button'>
```
Comment (YauhenKoipish) [inline src/components/Buttons/Hamburger/Hamburger.jsx:12]: <button onClick={toggle} className={showDrawer? 'bt-menu-trigger bt-menu-open' : 'bt-menu-trigger';} type='button'>
use ternary syntax better

File: src/components/Buttons/Hamburger/Hamburger.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  );
+}
+
+Hamburger.propTypes = {
+  showDrawer: PropTypes.bool,
+  toggle: PropTypes.bool,
+};
+
+Hamburger.defaultProps = {
+  showDrawer: PropTypes.bool,
```
Comment (YauhenKoipish) [inline src/components/Buttons/Hamburger/Hamburger.jsx:24]: use default props value

File: src/components/Drawer/Drawer.css [hunk 4]
```diff
@@ -0,0 +1,36 @@
+.side-drawer {
+  height: 100%;
+  color: #008f02;
```
Comment (YauhenKoipish) [inline src/components/Drawer/Drawer.css:3]: use variables color

File: src/components/Drawer/Drawer.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+export function Drawer(props) {
+  let drawerClasses = 'side-drawer';
+  const { showDrawer, toggle } = props;
+
+  if (showDrawer) {
+    drawerClasses = 'side-drawer open';
+  }
+
+  return (
+    <aside className={drawerClasses}>
```
Comment (YauhenKoipish) [inline src/components/Drawer/Drawer.jsx:24]: use ternary syntax better

File: src/components/Drawer/Drawer.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      <img className='dev-logo' src={devLogo} alt='dev-incubator-logo' />
+      {menuItems.map((item) => (
```
Comment (YauhenKoipish) [inline src/components/Drawer/Drawer.jsx:26]: use destructurization

File: src/pages/Members.jsx [hunk 7]
```diff
@@ -1,4 +1,90 @@
...
+        <span>{`(${membersBody.length})`}</span>
+      </p>
+      <table className='table'>
+        <thead className='table-head'>
+          {menuItems.map((item) => (
+            <th>{item}</th>
+          ))}
+        </thead>
+        <tbody className='table-body'>
+          {membersBody.map((item) => (
```
Comment (YauhenKoipish) [inline src/pages/Members.jsx:57]: use destructurization

File: src/pages/Progress.jsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        <span>{`(${ProgressBody.length})`}</span>
+      </p>
+      <table className='table'>
+        <thead className='table-head'>
+          {menuItems.map((item) => (
+            <th>{item}</th>
+          ))}
+        </thead>
+        <tbody className='table-body'>
+          {ProgressBody.map((item) => (
```
Comment (YauhenKoipish) [inline src/pages/Progress.jsx:55]: parse item

File: src/pages/Tasks.jsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        <span>{`(${TasksBody.length})`}</span>
+      </p>
+      <table className='table'>
+        <thead className='table-head'>
+          {menuItems.map((item) => (
+            <th>{item}</th>
+          ))}
+        </thead>
+        <tbody className='table-body'>
+          {TasksBody.map((item) => (
```
Comment (YauhenKoipish) [inline src/pages/Tasks.jsx:51]: parse item

File: src/components/Buttons/Button/Button.jsx (hunk 1 above)
Comment (Larkgronick) [reply src/components/Buttons/Button/Button.jsx]: Replaced.

File: src/components/Buttons/Hamburger/Hamburger.jsx (hunk 2 above)
Comment (Larkgronick) [reply src/components/Buttons/Hamburger/Hamburger.jsx]: Replaced.

File: src/components/Buttons/Hamburger/Hamburger.jsx (hunk 3 above)
Comment (Larkgronick) [reply src/components/Buttons/Hamburger/Hamburger.jsx]: Changed.

File: src/components/Drawer/Drawer.css (hunk 4 above)
Comment (Larkgronick) [reply src/components/Drawer/Drawer.css]: Replaced in all css files.

File: src/components/Drawer/Drawer.jsx (hunk 5 above)
Comment (Larkgronick) [reply src/components/Drawer/Drawer.jsx]: Replaced.

File: src/components/Drawer/Drawer.jsx (hunk 6 above)
Comment (Larkgronick) [reply src/components/Drawer/Drawer.jsx:19]: Changed.

File: src/pages/Members.jsx (hunk 7 above)
Comment (Larkgronick) [reply src/pages/Members.jsx:53]: Changed.

**Review by YauhenKoipish — APPROVED** [review]

**Pull request closed without merging**